Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,784 @@
+    {
+        return null;
+    }
+
+    /// <summary>
+    /// Writes the utility to XML.
+    /// </summary>
+    /// <param name="writer">The XML writer to write to.</param>
+    public void WriteXml(XmlWriter writer)
+    {
+        writer.WriteAttributeString("X", Tile.X.ToString());
+        writer.WriteAttributeString("Y", Tile.Y.ToString());
+        writer.WriteAttributeString("Z", Tile.Z.ToString());
+        writer.WriteAttributeString("objectType", ObjectType);
+
+        // Let the Parameters handle their own xml
+        Parameters.WriteXml(writer);
+    }
+
+    /// <summary>
+    /// Reads the prototype utility from XML.
+    /// </summary>
+    /// <param name="readerParent">The XML reader to read from.</param>
+    public void ReadXmlPrototype(XmlReader readerParent)
+    {
+        ObjectType = readerParent.GetAttribute("objectType");
+
+        XmlReader reader = readerParent.ReadSubtree();
+
+        while (reader.Read())
+        {
+            switch (reader.Name)
+            {
+                case "Name":
+                    reader.Read();
+                    Name = reader.ReadContentAsString();
+                    break;
+                case "TypeTag":
+                    reader.Read();
+                    typeTags.Add(reader.ReadContentAsString());
+                    break;
+                case "Description":
+                    reader.Read();
+                    description = reader.ReadContentAsString();
+                    break;
+                case "BuildingJob":
+                    float jobTime = float.Parse(reader.GetAttribute("jobTime"));
+
+                    List<Inventory> invs = new List<Inventory>();
+
+                    XmlReader inventoryReader = reader.ReadSubtree();
+
+                    while (inventoryReader.Read())
+                    {
+                        if (inventoryReader.Name == "Inventory")
+                        {
+                            // Found an inventory requirement, so add it to the list!
+                            invs.Add(new Inventory(
+                                    inventoryReader.GetAttribute("objectType"),
+                                    int.Parse(inventoryReader.GetAttribute("amount")),
+                                    0));
+                        }
+                    }
+
+                    Job j = new Job(
+                                null,
+                                ObjectType,
+                                FunctionsManager.JobComplete_UtilityBuilding,
+                                jobTime,
+                                invs.ToArray(),
+                                Job.JobPriority.High);
+                    j.JobDescription = "job_build_" + ObjectType + "_desc";
+                    PrototypeManager.UtilityJob.Set(ObjectType, j);
+                    break;
+
+                case "CanBeBuiltOn":
+                    TileType tileType = TileType.GetTileType(reader.GetAttribute("tileType"));
```

Comment:
This doesn't work since the TileTypes are loaded after the utilities prototypes. Plus we shouldn't asume any loading order. You can just add the tile type as a string. I have a PR to fix it for furniture: #1235